### DISCUSSION TITLE
What is the mathematic equation behind the model?

### DISCUSSION THREAD

**Discussion by hoangthangta** [Blealtan/efficient-kan#34] [General]
I wonder what the equation used in the KAN model, anybody knows?

**Comment by karlo-con-k** [comment]
From what I understand, conventional neural network models are based on the Universal Approximation Theorem. However, KAN is based on the Kolmogorov-Arnol Representation Theorem. I'm not sure if that answers your question :D

**Comment by hoangthangta** [comment]
We may know that KAN uses math equations (instead of variables) to fit the data points. However, what type of math equation (abs, sin, cos, etc) does this package use? It may use curve2coeff (compute the coefficients of the curve that interpolates the given points) to solve the equation (numpy.linalg.lstsq, a @ x = b). I am not sure but it is correct since the authors does not give much information about math equations.

**Comment by minh-nguyenhoang** [comment]
I think the author still use b-spline, but just compute them in a more efficient way.

> However, all activation functions are linear combination of a fixed set of basis functions which are B-splines; given that, we can reformulate the computation as activate the input with different basis functions and then combine them linearly.

**Comment by Blealtan** [comment]
Please refer to the original paper. I'm just reproducing it in a more computationally efficient way. The only change is in the sparsification loss term which seems not really critical to applications other than symbolic regression; if I get the time I will write on that one.

**Reply by EladWarshawsky** [reply]
I personally would be overjoyed if you added symbolic regression functionality
